templates/gitlab-manager-runner.yaml

```diff
@@ -0,0 +1,561 @@
+  VpcId:
+    Type: AWS::EC2::VPC::Id
+    Description: A VPC to run ECS instances in.
+  PrivateSubnetIds:
+    Type: List<AWS::EC2::Subnet::Id>
+    Description: Private subnet IDs from the VPC for the manager instance. Minimum of two.
+  DbHostProd:
+    Type: String
+    Description: The endpoint for the DB cluster for production. For example - mystack-mydbcluster-123456789012.us-east-2.rds.amazonaws.com
+    Default: ''
+    NoEcho: true
+  DbHostStaging:
+    Type: String
+    Description: The endpoint for the DB cluster for staging. For example - mystack-mydbcluster-123456789012.us-east-2.rds.amazonaws.com
+    Default: ''
+    NoEcho: true
+  DbPortProd:
+    Type: Number
+    Description: The port of the DB cluster for production.
+    Default: 5432
+    NoEcho: true
+  DbPortStaging:
+    Type: Number
+    Description: The port of the DB cluster for staging.
+    Default: 5432
+    NoEcho: true
+  DbCredsArnProd:
+    Description: The name of the Database secret in the Secret Manager for Prod.
+    NoEcho: 'true'
+    Type: String
+  DbCredsArnStaging:
+    Description: The name of the Database secret in the Secret Manager for Staging.
+    NoEcho: 'true'
+    Type: String
+  DBAccessSecurityGroupNameProd:
+    Type: String
+    Description: The DB Security Group's name for production.
+  DBAccessSecurityGroupNameStaging:
+    Type: String
+    Description: The DB Security Group's name for staging.
+  KmsKeyArnProd:
+    Description: 'Encryption key ARN to use for storage in Prod'
+    Type: String
+  KmsKeyArnStaging:
+    Description: 'Encryption key ARN to use for storage in Staging'
+    Type: String
+
+Conditions:
+  UseSpotInstances: !Equals ['Yes', 'Never']
+
+Resources:
+  ########################
+  ### Managed Policies ###
+  ########################
+  CachePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Action:
+              - 's3:ListObjects*'
+              - 's3:GetObject*'
+              - 's3:DeleteObject*'
+              - 's3:PutObject*'
+            Resource:
+              - !Sub '${CacheBucket.Arn}/*'
+          - Effect: 'Allow'
+            Action:
+              - 's3:ListBucket'
+              - 's3:AbortMultipartUpload'
+              - 's3:ListMultipartUploadParts'
+              - 's3:ListBucketMultipartUploads'
+            Resource:
+              - !GetAtt 'CacheBucket.Arn'
+  RunnerPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Action:
+              - 'ec2:*'
+            Resource: '*'
+            Condition:
+              StringEqualsIfExists:
+                'ec2:Region': !Ref 'AWS::Region'
+              ArnEqualsIfExists:
+                'ec2:Vpc': !Sub 'arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:vpc/${VpcId}'
```

Comment:
The idea here is that the policy would allow ec2:*/* on any ec2 in the minivpc?

Reply:
The intention is to limit the permissions to within the region and to only within the VPC that is created for this app (and runner).